drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,494 @@
+ *                  Example: -1 - I2C peripheral wasn't initialized or the
+ *                                device is not present.
+ *                            0 - I2C peripheral was initialized and the
+ *                                device is present.
+*******************************************************************************/
+int max9611_init(struct max9611_dev **device,
+		 struct max9611_init_param init_param)
+{
+	int ret;
+	struct max9611_dev *dev;
+
+	dev = (struct max9611_dev *)no_os_calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_i2c_init(&dev->i2c_desc, &init_param.i2c_init);
+	if (ret)
+		goto error_dev;
+
+	*device = dev;
+
+	return 0;
+
+error_dev:
+	no_os_free(dev);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Free the resources allocated by max538x_init().
+ * @param dev - The device structure.
+ * @return ret - The result of the remove procedure.
+*******************************************************************************/
+int max9611_remove(struct max9611_dev *dev)
+{
+	int ret;
+
+	ret = no_os_i2c_remove(dev->i2c_desc);
+	no_os_free(dev);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Reads data from device register (I2C)
+ * @param dev       - The device structure.
+ * @param addr      - Starting / selected register address.
+ * @param bytes     - Number of bytes to read from starting register address
+ * @param read_data - Read data buffer
+ * @return          - 0 if I2C read is successful, error otherwise.
+*******************************************************************************/
+int max9611_read(struct max9611_dev *dev, uint8_t addr,
+		 uint8_t bytes, uint8_t *read_data)
+{
+	int ret;
+
+	/* Check if bytes is 0 */
+	if (bytes <= 0)
+		return -EINVAL;
+
+	/* Check if valid register */
+	if (addr < MAX9611_CSA_MSB || addr > MAX9611_CTR2)
+		return -EINVAL;
+
+	/* Avoid read to out of bounds register */
+	if (addr + bytes > MAX9611_OUT_OF_BOUNDS)
+		return -EINVAL;
+
+	ret = no_os_i2c_write(dev->i2c_desc, &addr, 1, 0);
+	if (ret)
+		return ret;
+
+	return no_os_i2c_read(dev->i2c_desc, read_data, bytes, 1);
+}
+
+/***************************************************************************//**
+ * @brief Writes data to device register (I2C)
+ * @param dev 		  - The device structure.
+ * @param addr 		  - Starting / selected register address.
+ * @param bytes 	  - Number of bytes to write from starting register address
+ * @param write_data  - Write data buffer
+ * @return            - 0 if I2C write is successful, error otherwise.
+*******************************************************************************/
+int max9611_write(struct max9611_dev *dev, uint8_t addr,
+		  uint8_t bytes, uint8_t *write_data)
+{
+	int i;
+	uint8_t buf[bytes + 1];
+
+	/* Check if bytes is 0 */
+	if (bytes <= 0)
```

Comment:
`bytes` is unsigned, therefore this check is redundant.

Reply:
removed in latest commit.